### PR TITLE
Rule to check for structNew and suggest using {} instead

### PR DESCRIPTION
**Pull request by justinmclean**

### CODE REVIEW

**Comment by ryaneberly** [comment]
@justinmclean 
I combined the structnew() and writedump() checks into a single generic FunctionXChecker with 2 parameterized definitions in cflint.definition.xml

**Comment by justinmclean** [comment]
Does it understand / check that write dump take a parameter, where structnew doesn't?

**Comment by justinmclean** [comment]
And I assume you've not merged in arraynew(1), as arraynew(1) can be replace d by [] but arraynew(2) can't.

File: src/main/java/com/cflint/plugins/core/FunctionXChecker.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	final String severity = "INFO";
+
+	@Override
+	public void expression(CFExpression expression, Context context,
+			BugList bugs) {
+		if (expression instanceof CFFunctionExpression) {
+			final String cfmlFunctionCheck = getParameter("functionName");
+
+			CFFunctionExpression functionExpression = (CFFunctionExpression)expression;
+			if(functionExpression.getName().equalsIgnoreCase(cfmlFunctionCheck)){
```
Comment (justinmclean) [inline src/main/java/com/cflint/plugins/core/FunctionXChecker.java]: may want to use cfmlFunctionCheck  + "(" so less chance of matching things that are not function calls?

**Comment by ryaneberly** [comment]
Right, I didn't use this strategy for arraynew(1). I was going to, but one of the tests failed :-).

cfmlFunctionCheck  is the entire name.  The grammar will not call it a function unless there is a parenthesis though.

**Pull request merged**